README.md

```diff
@@ -336,6 +336,135 @@ from ft6x36 import ft6x36
+
+### Display orientation
+
+The `rot` parameter is used to set the MADCTL register of the display. The MADCTL register controls
+the order that pixels are written to the framebuffer. This sets the Orientation or Rotation of the
+display.
+
+See the [README.md](examples/madctl/README.md) file in the [examples/madctl](examples/madctl/)
+directory for more information on the MADCTL register and how to determine the `colormode` and `rot`
+parameters for a display.
+
+
+By default, the st7789 driver is initialized with the following parameters that are compatible with the TTGO T-Display:
+
+```
+    st7789(
+        miso=-1, mosi=19, clk=18, cs=5, dc=16, rst=23, power=-1, backlight=4,
+        backlight_on=1, power_on=0, spihost=esp.HSPI_HOST, mhz=40, factor=4, hybrid=True,
+        width=320, height=240, start_x=0, start_y=0, colormode=COLOR_MODE_BGR, rot=PORTRAIT,
+        invert=True, double_buffer=True, half_duplex=True, asynchronous=False, initialize=True)
+
+```
+
+  Arg | Description
+  --- | -----------
+  miso | Pin for SPI Data from display, -1 if not used as many st7789 displays do not have this pin
+  mosi | Pin for SPI Data to display (REQUIRED)
+  clk | Pin for SPI Clock (REQUIRED)
+  cs | Pin for display CS
+  dc | Pin for display DC (REQUIRED)
+  rst | Pin for display RESET
+  power | Pin for display Power ON, -1 if not used
+  power_on | Pin value for Power ON
+  backlight | Pin for display backlight control
+  backlight_on | Pin value for backlight on
+  spihost | ESP SPI Port
+  mhz | SPI baud rate in mhz
+  factor | Decrease frame buffer by factor
+  hybrid | Boolean, True to use C refresh routine, False for pure Python driver
+  width | Display width
+  height | Display height
+  colormode | Display colormode
+  rot | Display orientation, PORTRAIT, LANDSCAPE, INVERSE_PORTRAIT, INVERSE_LANDSCAPE or Raw MADCTL value that will be OR'ed with colormode
+  invert | Display invert colors setting
+  double_buffer | Boolean, True to use double buffering, False to use single buffer (saves memory)
+  half_duplex | Boolean, True to use half duplex SPI communications
+  asynchronous | Boolean, True to use asynchronous routines
+  initialize | Boolean, True to initialize display
+
+#### TTGO T-Display st7789 Configuration example
+
+```
+import lvgl as lv
+from ili9XXX import st7789
+
+disp = st7789(width=135, height=240, rot=st7789.LANDSCAPE)
+```
+
+#### TTGO TWatch-2020 st7789 Configuration example
+
+```
+import lvgl as lv
+from ili9XXX import st7789
+
+import axp202c
+
+# init power manager, set backlight
+axp = axp202c.PMU()
+axp.enablePower(axp202c.AXP202_LDO2)
+axp.setLDO2Voltage(2800)
+
+# init display
+disp = st7789(
+    mosi=19, clk=18, cs=5, dc=27, rst=-1, backlight=12, power=-1,
+    width=240, height=240, rot=st7789.INVERSE_PORTRAIT, factor=4)
+```
+### ST7735 128x128 Configuration Example
+
+```
+from ili9XXX import ili9341, MADCTL_MX, MADCTL_MY
+
+disp = ili9341(
```

Comment:
It's a bit confusing here, creating instances of `ili9341` for ST7735 displays, users might think this is a mistake.  

As you mentioned, I also think it would be a good idea to derive new classes (`st7735_128x128`, `st7735_128x160`) from `ili9XXX` (or even from `ili9341`) with some of the arguments set to the right default values.

Reply:
I agree it looks confusing and an st7735 class would be clearer.

Reply:
I noticed you just added `st7735` class.
Worth fixing the README.